src/tvheadend/ChannelTuningPredictor.cpp

```diff
@@ -74,7 +74,7 @@ uint32_t ChannelTuningPredictor::PredictNextChannelId(uint32_t tuningFrom, uint3
    * it will point at the channel we should tune to */
   std::set<ChannelPair>::iterator predictedIt = m_channels.cend();
 
-  if (std::next(fromIt, 1) == toIt || toIt->second == firstNum) {
+  if (fromIt == m_channels.cend() || std::next(fromIt, 1) == toIt || toIt->second == firstNum) {
```

Comment:
You're absolutely right, it seems the code runs into undefined behavior without this patch (advancing an iterator past the end). It seems to work on Linux and Windows though since no one has reported this yet.

@ksooo we'll have to backport this to the Jarvis branch.